From cac6af664619f9bdb65f7a1589b1b355b9c42d8e Mon Sep 17 00:00:00 2001
From: Guillaume Betous <guillaumex.betous@intel.com>
Date: Fri, 29 Nov 2013 17:22:36 +0100
Subject: [PATCH 8/8] Correct potential null pointer dereference

BZ: 155384

When fp is null, it can't be closed.

Change-Id: Id7ccb3428e0e1030110ffe9cae41110743b05b79
Category: aosp improvement
Domain: PUPDR-Framework-core
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Guillaume Betous <guillaumex.betous@intel.com>
---
 mkbootimg/mkbootimg.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mkbootimg/mkbootimg.c b/mkbootimg/mkbootimg.c
index 7b4ac0e..06a8408 100644
--- a/mkbootimg/mkbootimg.c
+++ b/mkbootimg/mkbootimg.c
@@ -189,7 +189,7 @@ static int read_write_data(void *outdata, size_t outsz,
         if(!fp) {
             perror("fdopen");
             result = 1;
-            goto rw_done;
+            goto exit;
         }
         remaining = insz;
         pos = indata;
@@ -219,6 +219,7 @@ static int read_write_data(void *outdata, size_t outsz,
 
 rw_done:
         fclose(fp);
+exit:
         return result;
     }
 }
-- 
1.7.10.4

